### DISCUSSION TITLE
The launcher has a bootstrap similar to original legacy launcher?

### DISCUSSION THREAD

**Discussion by Adahel** [olauncher/olauncher#45] [Q&A] [answered]
Good day,

I just got to know the project and a question soon arose, the launcher has a bootstrap
similar to original legacy launcher?

I don't think it's a good practice for the user to have to guess the day a new version will be released and have to manually download the launcher.

I also don't think it's a good practice not to have the launcher packaged for the specific platform (exe and dmg). The popular launchers I know do this to at least warn about the need for java for the lay user, when in practice for an advanced user it makes no difference.

If a stable bootstrap is done, I can also permanently replace the jar file inside the original legacy launcher exe on the Windows platform, for purely aesthetic and nostalgic reasons.

A small tutorial in case someone wants to modify the original legacy launcher exe:

1 - Extract the original minecraft.exe launcher stub using 7zip (open archive -> # -> 1.exe) or a hex editor.

2 - Combining the stub with the jar with the following command on Windows:

`copy file1/b+file2/b file3/b`

Example:

`copy minecraft_stub.exe/b+olauncher-1.7.2_09-redist.jar/b minecraft.exe/b`

Sorry for the large text and thanks for reading.

**Comment by bigfoot547** [comment, marked as answer]
>the launcher has a bootstrap similar to original legacy launcher?

depending on what you're referring to, yes and no. The "redistributable" jar is actually a small program that downloads the original launcher and patches it. This is to alleviate legal issues with distributing Mojang code. The diff (theoretically) contains olauncher's work only. Once the launcher is opened, however, the rest of the code is effectively unmodified (excluding the MSA stuff of course - this is where the major behavior changes in the launcher can be seen).

>I don't think it's a good practice for the user to have to guess the day a new version will be released and have to manually download the launcher.

Luckily, they don't. olauncher checks the releases page on this github repo occasionally for new updates, and notifies users. Try downloading an older version, and you'll see what I am referring to.

>I also don't think it's a good practice not to have the launcher packaged for the specific platform (exe and dmg). <snip>

You are entitled to that opinion. To me, the target audience of this launcher is people who feel nostalgia for the old launcher, or prefer it for performance/support reasons. These people are necessarily advanced users, or at least familiar with the idea of executing jar files (they probably used the official java launcher back in the day). Naturally, there will be people who slip through the cracks in this regard, but they present seldom enough that it is not worth the maintenance effort of distributing native packages. Though there has been some independent work on MacOS packages - join the discord for that.

>If a stable bootstrap is done, I can also permanently replace the jar file inside the original legacy launcher exe on the Windows platform, for purely aesthetic and nostalgic reasons.

The jar file that you have to download and run will always be different between updates. The lack of automatic, unattended updates is by design, for privacy reasons. Therefore, each update, you will need to regenerate this executable. Not a task that some kind of automation can't solve :)

**Reply by Adahel** [reply]
@bigfoot547 

> The lack of automatic, unattended updates is by design, for privacy reasons.

Good day,

According to my limited knowledge:

I didn't quite understand the last point. My interpretation is that you don't want to host files on your own domain lest you have data leaked via whois or a surprise dmca email, but this interpretation is probably wrong as you or anyone else can download the latest release of the file via the github API:

`https://api.github.com/repos/olauncher/olauncher/releases/latest`

In the link there is a json file, inside the `assets` array, each file added in the last release has a string `browser_download_url` that downloads the file.

Due to this possibility, I did not understand the reason for the lack of automatic.

Thanks,
Adahel.

**Reply by bigfoot547** [reply]
The "privacy" I was referring to was the privacy of the users, not my own privacy. A suspicious user might like the idea of OLauncher, but not trust that I won't silently release a nefarious update in the future. People have already had such concerns in the past. The only way I would be comfortable handling automatic updates would be if there were more maintainers of the project, and we all had to sign the update jar. For now, it is left up to the user to audit new versions.

For how infrequently major updates of olauncher are released, I think manually downloading new versions is not that much of an inconvenience.

**Comment by bigfoot547** [comment]
Also I'll convert this issue into a discussion because it is not a bug report or feature request.

**Comment by rabiesland** [comment]
if you really want an automatic launcher:
https://gist.github.com/unlikethesun/2d581bad39b9f9f555cde731acaaa339
